src/features/runtime-checks.js

```diff
@@ -497,133 +475,6 @@ function isRuntimeElement (element) {
     return false
 }
 
-function overloadGetOwnPropertyDescriptor () {
```

Comment:
Moving code around like this really belongs in it's own refactoring PR. The diff here would be much easier to understand passing in the `ContentFeature` `this`.

The rationale of not having these within the class is to prevent usages of `this` given the nested nature of them.

I'd prefer for now we just revert these changes specifically because it's a feature not in active development with perhaps less tests we'd like.

Moving big blocks of code around IMO should be avoided within some other piece of work. The time to review these meaningfully is very costly to ensure there is no shadowing and the code has correctly moved across.

As a quicker fix just set a file scope global once the init/load method is fired and wire through the `ContentFeature` `this`.

Reply:
I disagree that moving code should be avoided. I personally much prefer these changes to changes to the API signatures and calling mechanisms. Those have longer lasting consequences (limit further code changes), look confusing for first-time readers, and are essentially a tech debt. Git(hub) is a powerful tool, one can always track changes down, no matter how big or old they are.
As for the review costs, I intentionally made this move in a separate commit, so you can exclude it from the diff.
That said, I agree with your argument about using `this` inside nested functions, and since this is the only blocking thing, I just made the change.